### DISCUSSION TITLE
How to pass the file from same repo to my code

### DISCUSSION THREAD

**Discussion by infosechris** [community/community#25717] [Actions]
<p>Hello,</p>
<p>If I have a text file stored in my repo at:<br>
<code>repo/stored_files/hello.txt</code></p>
<ol>
<li>How do I pass this file into <code>mycode.py</code>? Path, variable.?</li>
<li>And once passed down, is it the .txt file that is passed down? or content in the .txt file as string? ( I want to able to use <code>with open (hello.txt, 'r')</code> as a file )</li>
<li>Is actions/checkout@v2 required? Can I just use <code>GITHUB_WORKSPACE</code> without checkout?</li>
<li>Does git actions automagically use its own auth from the workflow? I didn’t pass down any gh API token to run this.</li>
</ol>
<p>Example below:</p>
<pre><code class="lang-auto">jobs:
  dothis:
    runs-on: ubuntu-latest
    steps: 
    - name: Checkout Repo
      uses: actions/checkout@v2

    - name: Setup Python
      uses: actions/setup-python@v2
      with:
        python-version: '3.10.2'

    - name: Install requirements
      run: |
        pip -m pip install --upgrade pip  
        pip install -r requirements.txt
    
    - name: Python
      run: |
        python3 mycode.py
</code></pre>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="infosechris" data-post="1" data-topic="225679">
<div class="title">
<div class="quote-controls"></div>
 infosechris:</div>
<blockquote>
<p>Is actions/checkout@v2 required? Can I just use <code>GITHUB_WORKSPACE</code> without checkout?</p>
</blockquote>
</aside>
<p><code>actions/checkout</code> is what gets your repository. If you don’t call it, the workspace is empty. :wink:</p>
<aside class="quote no-group" data-username="infosechris" data-post="1" data-topic="225679">
<div class="title">
<div class="quote-controls"></div>
 infosechris:</div>
<blockquote>
<p>Does git actions automagically use its own auth from the workflow? I didn’t pass down any gh API token to run this.</p>
</blockquote>
</aside>
<p>Yes, <a href="https://docs.github.com/en/actions/security-guides/automatic-token-authentication">the <code>GITHUB_TOKEN</code></a>. <code>actions/checkout</code> uses it automatically if you don’t give it any other token.</p>
<aside class="quote no-group" data-username="infosechris" data-post="1" data-topic="225679">
<div class="title">
<div class="quote-controls"></div>
 infosechris:</div>
<blockquote>
<p>How do I pass this file into <code>mycode.py</code> ? Path, variable.?</p>
</blockquote>
</aside>
<p>After checkout the file exists in the workspace, just like in your repository. You can <code>open()</code> it, or read it any other way that makes sense for what you want to do.</p>

**Comment by infosechris** [comment]
<p>Thank you <a class="mention" href="/airtower-luna">@airtower-luna</a> ! This is very helpful.</p>
<p>How do I pass the path into the <code>mycode.py</code>?<br>
Will it be <code>env: repo/stored_files/hello.txt</code> under Python step? Is that path syntax correct?</p>

**Comment by airtower-luna** [comment]
<p>That depends on how your script expects to retrieve the path. The easiest thing would probably be to put the path on the command line, but you could also set an environment variable.</p>

**Comment by infosechris** [comment]
<p>So technically python is running on same path as the repo, I can just call it with <code>/</code> pathing?</p>
<p>repo/stored_files/hello.txt?</p>
<p>Can I call this path inside <code>.py</code> without passing it down as env variable?<br>
Or does it need to be specified in .yml and have it pass down with env variable?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The default working directory is <em>inside</em> your repository, so it’s going to be more like <code>stored_files/hello.txt</code>. You can add a step with <code>run: find .</code> to see what’s in the workspace.</p>
<p>The rest depends on how you <em>want</em> to do things. You <em>can</em> hardcode the path in your script. Whether that’s a good idea depends on what your script is supposed to do.</p>

**Comment by infosechris** [comment]
<p>Thank you <a class="mention" href="/airtower-luna">@airtower-luna</a> ! Much appreciate your help and insight!</p>
